src_tracer/database.py

```diff
@@ -0,0 +1,70 @@
+import logging
+import sqlite3
+import os
+
+from typing import List, Tuple
+
+log = logging.getLogger(__name__)
+
+class Database:
+
+    def __init__(self, store_dir, name="function_database.db", path=None):
+        if path:
+            self.connection = sqlite3.connect(path)
+        else:
+            db_path = os.path.join(store_dir, name)
+            self.connection = sqlite3.connect(db_path)
+
+    def get_name(self, func_num):
+        cursor = self.connection.cursor()
+        cursor.execute("SELECT name FROM function_list WHERE rowid=?", (func_num,))
+        name = cursor.fetchone()
+        if name is None:
+            log.error("there is no name (entry) for the func_num (%d)", func_num)
+            return None
+        cursor.close()
+        return name[0]
+
+    def get_numbers(self, func_name):
+        cursor = self.connection.cursor()
+        cursor.execute("SELECT rowid FROM function_list WHERE name=?", (func_name,))
+        func_nums = [func_num[0] for func_num in cursor.fetchall()]
+        cursor.close()
+        return func_nums
+
+    def create_table(self, table_name: str, colum_name_type: List[Tuple[str, str]], key: List[str]):
+        cursor = self.connection.cursor()
+        columns = ", ".join([" ".join(col) for col in colum_name_type])
+        key_str = ", ".join(key)
+        cursor.execute(f'''
+                       CREATE TABLE IF NOT EXISTS {table_name}
+                            ( {columns}, PRIMARY KEY ({key_str}) )
+                       ''')
+        cursor.close()
```

Comment:
This does not need to be public -- only for use in the Database class, so rename to `_create_table`.